src/routes.rs

```diff
@@ -227,8 +251,72 @@ pub async fn index(request: HttpRequest, identity: Identity) -> impl Responder {
             INDEX_CONTENTS.to_owned()
         }
     };
+    let final_contents = contents.replace("{{ themepath }}", &CONFIG.highlight.themepath);
     HttpResponse::Ok()
         .header("Content-Type", "text/html")
+        .body(final_contents)
+}
+
+/// CSS file to style the page from a local source
+pub async fn css(request: HttpRequest, identity: Identity) -> impl Responder {
+    if let Err(response) = auth(identity, request).await {
+        return response;
+    }
+
+    let contents = {
+        #[cfg(feature = "dev")]
+        {
+            fs::read_to_string(&*CSS_PATH).expect("Can't read spectre.min.css")
+        }
+        #[cfg(not(feature = "dev"))]
+        {
+            CSS_CONTENTS.to_owned()
+        }
+    };
+    HttpResponse::Ok()
+        .header("Content-Type", "text/css")
+        .body(contents)
+}
+
+/// Icon-specific CSS file to style the page from a local source
+pub async fn icon(request: HttpRequest, identity: Identity) -> impl Responder {
+    if let Err(response) = auth(identity, request).await {
+        return response;
+    }
+
+    let contents = {
+        #[cfg(feature = "dev")]
+        {
+            fs::read_to_string(&*ICON_PATH).expect("Can't read spectre.min.css")
+        }
+        #[cfg(not(feature = "dev"))]
+        {
+            ICON_CONTENTS.to_owned()
+        }
+    };
+    HttpResponse::Ok()
+        .header("Content-Type", "text/css")
+        .body(contents)
+}
+
+/// JS file to highlight code from a local source
+pub async fn js(request: HttpRequest, identity: Identity) -> impl Responder {
+    if let Err(response) = auth(identity, request).await {
+        return response;
+    }
+
+    let contents = {
+        #[cfg(feature = "dev")]
+        {
+            fs::read_to_string(&*JS_PATH).expect("Can't read spectre.min.css")
+        }
+        #[cfg(not(feature = "dev"))]
+        {
+            JS_CONTENTS.to_owned()
+        }
+    };
+    HttpResponse::Ok()
+        .header("Content-Type", "text/javascript")
         .body(contents)
 }
```

Comment:
Same here, a lot of duplicate code that could be merged, however that one is trickier since different values are used depending on whether or not the dev feature is enabled. Cleanest way to do it would probably be a macro, I can take care of that since you're new to Rust.